x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/CacheService.java

```diff
@@ -52,40 +69,65 @@
         Setting.Property.NodeScope
     );
 
+    public static final TimeValue MIN_SNAPSHOT_CACHE_SYNC_INTERVAL = TimeValue.timeValueSeconds(10L);
+    public static final Setting<TimeValue> SNAPSHOT_CACHE_SYNC_INTERVAL_SETTING = Setting.timeSetting(
+        SETTINGS_PREFIX + "sync_interval",
+        TimeValue.timeValueSeconds(60L),                        // default
+        MIN_SNAPSHOT_CACHE_SYNC_INTERVAL,                       // min
+        Setting.Property.NodeScope,
+        Setting.Property.Dynamic
+    );
+
+    private static final Supplier<Set<CacheFile>> SUPPLIER_OF_CACHE_FILES_TO_SYNC = ConcurrentCollections::newConcurrentSet;
```

Comment:
Looks like this might as well be a method (or inlined)?

Reply:
I removed this when moving to a Queue implementation.